server/handlers/attachments.go

```diff
@@ -226,9 +226,16 @@ func (ah *ActivityHandler) GetFileFromTeamsAndUploadToMM(downloadURL string, cli
 		return ""
 	}
 
-	recovery.Go("get_file_content_stream", ah.plugin.GetAPI().LogError, func() {
+	go func() {
+		defer func() {
```

Comment:
We could move this inside the client and return an error.

Reply:
This one I'm less convinced about, as the decision to run in a goroutine is the caller's. Would I have to add `recover()` to all client methods in case the caller decides to invoke in a goroutine?

Reply:
Your right, you could pass a pipe that has a goroutine waiting on read. Then the parent goroutine would handle recovery (or not :P)